eduHub/src/main/kotlin/com/arsiu/eduhub/grpc/service/AssignmentServiceGrpcImpl.kt

```diff
@@ -22,17 +24,28 @@ class AssignmentServiceGrpcImpl(
     private val findAllStreamHandler: FindAllStreamHandler,
     private val findByIdHandler: FindByIdHandler,
     private val updateHandler: UpdateHandler,
-    private val deleteByIdHandler: DeleteByIdHandler
+    private val deleteByIdHandler: DeleteByIdHandler,
+    val sharedAssignmentStream: SharedAssignmentStream,
+    val producer: AssignmentKafkaProducer
 ) : AssignmentServiceImplBase() {
 
     override fun findAll(request: Mono<FindAllAssignmentRequest>): Flux<FindAllAssignmentStreamResponse> =
-        request.flatMapMany { findAllStreamHandler.handleFindAll(it) }
+        Flux.concat(
+            request.flatMapMany { findAllStreamHandler.handleFindAll(it) },
+            sharedAssignmentStream.flux
+        ).log()
+
 
     override fun findById(request: Mono<FindByIdAssignmentRequest>): Mono<FindByIdAssignmentResponse> =
         request.flatMap { findByIdHandler.handleFindById(it) }
 
     override fun update(request: Mono<UpdateAssignmentRequest>): Mono<UpdateAssignmentResponse> =
-        request.flatMap { updateHandler.handleUpdate(it) }
+        request.flatMap { updateRequest ->
+            updateHandler.handleUpdate(updateRequest)
+                .doOnNext { response ->
+                    producer.sendAssignmentUpdateToKafka(response.response.success.assignment)
+                }
+        }
```

Comment:
```suggestion
        request.flatMap { updateRequest ->
            updateHandler.handleUpdate(updateRequest)
        }
         .doOnNext {
                    producer.sendAssignmentUpdateToKafka(it.response.success.assignment)
                }
```
please avoid nested reactor chains for better readability

Reply:
Understood